featurebyte/query_graph/enum.py

```diff
@@ -96,6 +96,18 @@ class NodeType(StrEnum):
     GRAPH = "graph"
     PROXY_INPUT = "proxy_input"
 
+    @classmethod
+    def inplace_operation_node_type(cls) -> Set["NodeType"]:
+        """
+        Returns all the node types that are inplace operations. Inplace operations are operations that
+        modify the input node inplace.
+
+        Returns
+        -------
+        Set[NodeType]
+        """
+        return {cls.ASSIGN, cls.CONDITIONAL, cls.ALIAS}
```

Comment:
What do you think of making this a node level attribute / property that each node has to define?

Reply:
I think that is even better as we force developer to think about this when inheriting the `BaseNode` class. Will move this to the node class.